### PR TITLE
adapt tests and TX timed outs

### PR DESCRIPTION
**Pull request by goncer**
* Fixed some left over from the old "waitForTx"
* getCurrentNonce now check if Sudo -> get from DB.
* only wait if force flag is set ( lets test removing the waiters a bit :) )

run 3 times, all good!

Test Suites: 18 passed, 18 total
Tests:       86 passed, 86 total
Snapshots:   0 total
Time:        782.074 s

### CODE REVIEW

File: utils/tx.ts [hunk 1]
```diff
@@ -98,14 +101,30 @@ export async function calculate_sell_price_id_rpc(soldTokenId: BN, boughtTokenId
...
+      const nonce = new BN(await SudoDB.getInstance().getSudoNonce(account));
+
+      return nonce;
+
+  }else if (account){
+
+    return getChainNonce(account);
+
   }
   return new BN(-1)
```
Comment (mateuszaaa) [inline utils/tx.ts:118]: i know that its not part of this PR but shouldnt some error be reported instead

File: utils/txHandler.ts [hunk 2]
```diff
@@ -139,77 +137,80 @@ export const signAndWaitTx = async (
...
+                            return;
+                        }
+                        
+    
+                        let req_events = curr_block_events.filter(event => {
+                            return event.phase.isApplyExtrinsic && event.phase.asApplyExtrinsic.toNumber() === index;
+                        }).map( ({ phase, event }) => {
+                            return event;
+                        });
+                        console.log(
```
Comment (mateuszaaa) [inline utils/txHandler.ts:184]: shouldnt that be enable on some env variable or switch

**Review by mateuszaaa — APPROVED** [review]
great to see that feature ready ! few minor comments

File: utils/tx.ts (hunk 1 above)
Comment (goncer) [reply utils/tx.ts:118]: AFIK ( an I am not 100% sure ) a nonce of -1 is a "default" value for nonce. 
But! if you are sure that is not valid, I'll trigger an exception instead, that would make more sense.

File: utils/txHandler.ts (hunk 2 above)
Comment (goncer) [reply utils/txHandler.ts:184]: Right! I'll review all the traces and handle extra debug options accordingly.

Comment (goncer) [reply utils/txHandler.ts]: Added a logging framework,
```
[2021-08-02T11:50:39.959Z] - W[3] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:39.949Z] - W[4] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.020Z] - W[1] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.001Z] - W[5] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.029Z] - W[6] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:39.990Z] - W[7] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.028Z] - W[2] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:41.935Z] - W[1] - [debug]: 0x41b2b6ce07a6aab7dd420715ab63ec1b8b06f17d7f8d67b5599e15cce53b4631
[2021-08-02T11:50:41.954Z] - W[1] - [info]: Chain is at block: #470
[2021-08-02T11:50:42.021Z] - W[1] - [info]: Chain is at block: #470
```
/mangata-e2e/utils/Logger.ts
* Which define custom levels based on the env.Variable.
* Write into a file ( that will be exported in the actions )
* Defined a custom formatting! 

```
Test Suites: 18 passed, 18 total
Tests:       86 passed, 86 total
Snapshots:   0 total
Time:        928.571 s
```
Thanks for the hint @mateuszaaa !
